README.md

```diff
@@ -19,7 +30,166 @@ After installation, you will need to install `react-navigation` and its peer dep
 
 ## Examples
 
-- example built-in library - [/example](/example)
+- example built-in library - check [/example](/example)
+
+## API
+
+For all navigator you need follow some steps:
+
+### Config
+
+First of all, create a config file like `Config.re` and there define your all routes. It sould be a simple Varian Type with your routes/tabs/items
+
+```ReasonML
+type route =
+  | Home
+  | UserDetails;
+```
+
+> It is important to create a separate file for that because ReasonML is aware of circular dependencies.
+
+### Navigation prop for compoenents
+
+For our components we need to create navigationProp type, which is created from a list of our routes defined in [Confgi.re](#Config).
+
+```ReasonML
+type navigationProp = StackNavigator.navigation(route);
+```
+
+> Each Navigator expose their own navitationProp type.
+
+Example:
+
+```ReasonML
+let make = (~navigation: Config.navigationProp, _children) => {
+  ...component,
+  render: _self =>
+    <View>
+      <Button
+        title="Go to home screen"
+        onPress={() => navigation.push(Home)}
+      />
+      <Button
+        title="Go back"
+        onPress={() => navigation.goBack()}
+      />
+    </View>,
+};
+```
+
+### StackNavigator
+
+#### Configuration
+
+Use a functor `Create` from `StackNavigator` module and pass a configuration module which needs to include a few pieces:
+
+- `route` - list of your routes, use variant from your `Config.re`
+- `initialRoute` - the first screen of your navigation state
+- `getScreen` - it's a function which as a parameters get the `currentRoute` and `navigation` props. As a return, you should create a tuple where the first element is a screen component and the second is screen configuration.
+
+#### Route Params
+
+If your route has a parameter you should pass them to you component inside `getScreen` function.
+
+exmaple:
+
+```ReasonML
+
+let getScreen = (currentRoute, navigation) =>
+  switch (currentRoute) {
+  | Home => (<Screen navigation />, screenOptions(~title="Home", ()))
+  | UserDetails(userId) => (
+      <Screen navigation text={"Browsing profile of: " ++ userId} />,
+      screenOptions(~title="Hello " ++ userId, ()),
+    )
+  };
+```
+
+### SwitchNavigator
+
+The API for creating navigator is almost the same as in Stack Navigator:
+
+```ReasonML
+module Switch =
+  SwitchNavigator.Create({
+    open SwitchNavigator;
+
+    type route = Config.route;
+
+    let initialRoute = Login;
+
+    let getScreen = (currentRoute, navigation) =>
+      switch (currentRoute) {
+      | Login => (<Login navigation />, screenOptions())
+      | LoggedIn => (<LoggedIn navigation />, screenOptions())
+      };
+  });
+```
+
+### TabNavigator
+
+Tab needs one additional setting compared to the Switch or Stack Navigator.
+
+This is list of items that drawer needs to render itself:
```

Comment:
Not sure if this sentence is correct?

Reply:
This is something that left from the previous docs - I'll remove it for now.